micro-rdk/src/common/config_monitor.rs

```diff
@@ -72,9 +78,54 @@ where
                 .await?;
 
             if new_config
+                .as_ref()
                 .config
-                .is_some_and(|cfg| cfg != *self.curr_config)
+                .as_ref()
+                .is_some_and(|cfg| *cfg != *self.curr_config)
             {
+                log::info!("new robot config detected");
+
+                #[cfg(feature = "ota")]
+                {
+                    let config = new_config.config.clone().unwrap();
+                    if let Some(service) = config
+                        .services
+                        .iter()
+                        .find(|&service| service.model == *ota::OTA_MODEL_TRIPLET)
+                    {
+                        match ota::OtaService::from_config(
+                            service,
+                            self.storage.clone(),
+                            self.executor.clone(),
+                        ) {
+                            Ok(mut ota) => {
+                                let curr_metadata = ota.stored_metadata().await;
+                                if curr_metadata.version != ota.pending_version {
+                                    log::info!(
+                                        "firmware is out of date, updating from `{}` to `{}`",
+                                        curr_metadata.version,
+                                        ota.pending_version
+                                    );
+                                    while let Err(ota_err) = ota.update().await {
+                                        log::error!("failed to update firmware: {}", ota_err);
+                                        log::info!("retrying firmware update");
+                                    }
```

Comment:
wasn't sure if I should put a delay here at all since there is one internally in `update` for connection establishment already.

Reply:
additionally, if the error is due to specific errors `ConfigError`, `InvalidImage`, etc, I'm not sure what the behavior should be.

we can 
a) continue applying the other configs (might be risky if the new configs are only compatible with new firmware)
b) continue to block on applying the current update and surface an error

Reply:
I think a) is the right move here. Consumers of the pre-packaged micro-rdk-server can see why the configs incompatible with the old firmware failed to function appropriately because the version on app will be the old version. Others can look at the logs (which is admittedly not ideal, but the solution to that is to change/revamp version reporting as we have previously discussed)